src/tir/transforms/merge_shared_memory_allocations.cc

```diff
@@ -156,29 +156,18 @@ class SharedMemLinearAccessPatternFinder final : public StmtExprVisitor {
     if (it != alloc_info_.end() && it->second.alloc) {
       ICHECK_LT(it->second.level, scope_.size()) << "Load memory in places other than store.";
       if (IsAppropriateSharedMemory(GetRef<Var>(buf))) {
-        scope_[it->second.level].touched.push_back(buf);
+        scope_[scope_.size() - 1].touched.push_back(buf);
       }
     }
   }
 
-  void VisitExpr_(const CallNode* op) final {
```

Comment:
remove this visit function, as it only allow visit indices, which will lead to some buffer load statement not be traced.